src/components/PeoplePage.tsx

```diff
@@ -1,8 +1,109 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { useEffect, useMemo, useState } from 'react';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
+import { SortTypes } from '../types/SortTypes';
+import { SexFilters } from '../types/SexFilters';
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+
+  function addParents(person: Person, peopleFromServer: Person[]) {
+    const mother = peopleFromServer.find(
+      pers => pers.name === person.motherName,
+    );
+    const father = peopleFromServer.find(
+      pers => pers.name === person.fatherName,
+    );
+
+    return { ...person, father, mother };
+  }
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(result => {
+        setPeople(result.map(person => addParents(person, result)));
+        setTimeout(() => {
+          setIsLoading(false);
+        }, 300);
+      })
+      .catch(() => {
+        setIsLoading(false);
+        setIsError(true);
+      });
+  }, []);
+
+  const visiblePeople = useMemo(() => {
+    let formattedPeople = [...people];
+
+    switch (searchParams.get('sort')) {
+      case SortTypes.Name:
+        formattedPeople.sort((a, b) => a.name.localeCompare(b.name));
+        break;
+
+      case SortTypes.Sex:
+        formattedPeople.sort((a, b) => a.sex.localeCompare(b.sex));
+        break;
```

Comment:
above: `const sortParam = searchParams.get('sort');`

```suggestion
      case SortTypes.Name:
      case SortTypes.Sex:
        formattedPeople.sort((a, b) => a[sortParam].localeCompare(b[sortParam]));
        break;
```
if logic is the same you can join some cases

Reply:
yes, I just forgot about this feature